core/services/keystore/csa.go

```diff
@@ -133,6 +135,28 @@ func (ks *csa) Export(id string, password string) ([]byte, error) {
 	return key.ToEncryptedJSON(password, ks.scryptParams)
 }
 
+// EnsureKey verifies whether the CSA key has been seeded, if not, it creates it.
+func (ks *csa) EnsureKey() error {
+	ks.lock.Lock()
+	defer ks.lock.Unlock()
+	if ks.isLocked() {
+		return ErrLocked
+	}
+
+	if len(ks.keyRing.CSA) > 0 {
+		return ErrCSAKeyExists
```

Comment:
In case a key already exists, it is better to just return nil here.
The function EnsureKey() clearly means that it should only ensure that key is there.
Error is only reserved for cases when it couldn't figure out the key status, or failed while creating a new key.
This way, on the caller side, you won't have to check for the error value too.